rakelib/plugin.rake

```diff
@@ -29,6 +32,40 @@ namespace "plugin" do
     LogStash::PluginManager::Main.run("bin/logstash-plugin", ["remove", plugin] + more_plugins)
   end
 
+  def list_plugins(search=nil, expand: nil, verbose: nil)
+    require_relative "../lib/pluginmanager/main"
+    args = []
+    args << "--verbose" if verbose
+    args << search unless search.nil?
+
+    stdout = invoke_plugin_manager!("list", *args)
+
+    stdout.lines.select do |line|
+      # STDOUT pollution removal needed until 8.19 and 9.1
+      # https://github.com/elastic/logstash/pull/17125
+      next false if line.match?(/^Using (system java|bundled JDK|LS_JAVA_HOME defined java)/)
+      
+      # post-execution filtration is needed until 8.19 and 9.1
+      # when list --[no-]expand flag is supported, use it instead
+      # https://github.com/elastic/logstash/pull/17124
+      expand || p.match?(/^[a-z]/)
+    end.map(&:chomp)
+  end
+
+  def clean_plugins
+    invoke_plugin_manager!("clean")
+  end
```

Comment:
✂️ the `clean` command was absorbed into the plugin manager's mutation commands.

```suggestion
```